components/editor/CodeBlock.tsx

```diff
@@ -0,0 +1,583 @@
+    color: '#74e287'
+  },
+  'hljs-keyword': {
+    color: '#64a0dc'
+  },
+  'hljs-literal': {
+    color: '#64a0dc'
+  },
+  'hljs-symbol': {
+    color: '#64a0dc'
+  },
+  'hljs-number': {
+    color: '#d8da68'
+  },
+  'hljs-regexp': {
+    color: '#EBCB8B'
+  },
+  'hljs-string': {
+    color: '#c0e2a3',
+    fontWeight: '500'
+  },
+  'hljs-title': {
+    color: '#7edcda'
+  },
+  'hljs-params': {
+    color: '#d6deeb'
+  },
+  'hljs-bullet': {
+    color: '#64a0dc'
+  },
+  'hljs-code': {
+    color: '#7edcda'
+  },
+  'hljs-emphasis': {
+    fontStyle: 'italic'
+  },
+  'hljs-formula': {
+    color: '#7edcda'
+  },
+  'hljs-strong': {
+    fontWeight: 'bold'
+  },
+  'hljs-link:hover': {
+    textDecoration: 'underline'
+  },
+  'hljs-quote': {
+    color: '#797f8c'
+  },
+  'hljs-comment': {
+    color: '#797f8c'
+  },
+  'hljs-doctag': {
+    color: '#7edcda'
+  },
+  'hljs-$ref': {
+    color: 'yellow'
+  },
+  'hljs-meta': {
+    color: '#5E81AC'
+  },
+  'hljs-meta-keyword': {
+    color: '#5E81AC'
+  },
+  'hljs-meta-string': {
+    color: '#c0e2a3'
+  },
+  'hljs-attr': {
+    color: '#7edcda'
+  },
+  'hljs-attribute': {
+    color: '#d6deeb'
+  },
+  'hljs-builtin-name': {
+    color: '#64a0dc'
+  },
+  'hljs-name': {
+    color: '#64a0dc'
+  },
+  'hljs-section': {
+    color: '#74e287'
+  },
+  'hljs-tag': {
+    color: '#64a0dc'
+  },
+  'hljs-variable': {
+    color: '#d6deeb'
+  },
+  'hljs-template-variable': {
+    color: '#d6deeb'
+  },
+  'hljs-template-tag': {
+    color: '#5E81AC'
+  },
+  'yaml .hljs-meta': {
+    color: '#D08770'
+  }
+};
+
+const CodeBlock: React.FC<CodeBlockProps> = ({
+  children,
+  codeBlocks,
+  className = '',
+  highlightClassName = '',
+  highlightedLines,
+  language = 'yaml',
+  hasWindow = false,
+  showCopy = true,
+  showCaption = true,
+  caption = '',
+  showLineNumbers = true,
+  startingLineNumber = 1,
+  textSizeClassName = 'text-xs',
+  title = language
+}) => {
+  const [activeBlock, setActiveBlock] = useState<number>(0);
+  const [showIsCopied, setShowIsCopied] = useState<boolean>(false);
+
+  // eslint-disable-next-line no-param-reassign
+  codeBlocks = codeBlocks && codeBlocks.length ? codeBlocks : [{ code: children.replace(/\n$/, '') }];
+
+  const tabItemsCommonClassNames =
+    'inline-block border-teal-300 py-1 px-2 mx-px cursor-pointer hover:text-teal-300 font-bold';
+  const tabItemsClassNames = `${tabItemsCommonClassNames} text-gray-300`;
+  const tabItemsActiveClassNames = `${tabItemsCommonClassNames} text-teal-300 border-b-2`;
+
+  function onClickCopy() {
+    // check if navigator with clipboard exists (fallback for older browsers)
+    if (navigator && navigator.clipboard && (codeBlocks && codeBlocks[activeBlock])) {
+      navigator.clipboard.writeText(codeBlocks[activeBlock].code).then(() => {
+        setShowIsCopied(true);
+        setTimeout(() => {
+          setShowIsCopied(false);
+        }, 2000);
+      });
+    }
+  }
+
+  function renderHighlight() {
```

Comment:
Provide jsdoc comments for these functions to get better understanding.

Reply:
Done